cassandra/src/main/scala/filodb.cassandra/columnstore/CassandraColumnStore.scala

```diff
@@ -215,6 +217,79 @@ extends ColumnStore with CassandraChunkSource with StrictLogging {
     }
   }
 
+  /**
+    * Copy a range of chunks to a target ColumnStore, for performing disaster recovery or
+    * backfills.
+    */
+  // scalastyle:off null method.length
+  def copyChunksByIngestionTimeRange(datasetRef: DatasetRef,
+                                     splits: Iterator[ScanSplit],
+                                     ingestionTimeStart: Long,
+                                     ingestionTimeEnd: Long,
+                                     batchSize: Int,
+                                     target: CassandraColumnStore,
+                                     targetDatasetRef: DatasetRef,
+                                     diskTimeToLiveSeconds: Int): Unit =
+  {
+    val sourceIndexTable = getOrCreateIngestionTimeIndexTable(datasetRef)
+    val sourceChunksTable = getOrCreateChunkTable(datasetRef)
+
+    val targetIndexTable = target.getOrCreateIngestionTimeIndexTable(targetDatasetRef)
+    val targetChunksTable = target.getOrCreateChunkTable(targetDatasetRef)
+
+    val chunkInfos = new ArrayBuffer[ByteBuffer]()
+    val futures = new ArrayBuffer[Future[Response]]()
+
+    def finishBatch(partition: ByteBuffer): Unit = {
+      for (row <- sourceChunksTable.readChunks(partition, chunkInfos).iterator.asScala) {
+        futures += targetChunksTable.writeChunks(partition, row, diskTimeToLiveSeconds)
+      }
+
+      chunkInfos.clear()
+
+      for (f <- futures) {
+        try {
+          Await.result(f, Duration(10, SECONDS))
+        } catch {
+          case e: Exception => {
+            logger.error(s"Async cassandra chunk copy failed", e)
+          }
+        }
+      }
+
+      futures.clear()
+    }
+
+    var lastPartition: ByteBuffer = null
+
+    for (split <- splits) {
+      val tokens = split.asInstanceOf[CassandraTokenRangeSplit].tokens
+      val rows = sourceIndexTable.scanRowsByIngestionTime(tokens, ingestionTimeStart, ingestionTimeEnd)
+      for (row <- rows) {
+        val partition = row.getBytes(0) // partition
```

Comment:
I am not so much liking the leak in abstraction. CassColStore is needing to know about which columns are selected by CQL issued in index table class. In this case, it is ok to pay cost of case class instead of using "Row".

Reply:
Not much of leak really. All the code in this package is altogether a "unit" and nothing leaks out. If the table classes were in the same file as the CassandraColumnStore it would look like less of a "leak", but not really. I can add an extra wrapper around the row (no need to copy, since it's just a wrapper), but it would just map method names to numbers.

Reply:
I should be able to declare these methods such that access is restricted just to CassandraColumnStore.